evalml/pipelines/components/transformers/transformer.py

```diff
@@ -54,9 +51,8 @@ def transform(self, X, y=None):
                 "Transformer requires a transform method or a component_obj that implements transform"
             )
         X_t_df = pd.DataFrame(X_t, columns=X_ww.columns, index=X_ww.index)
-        return _retain_custom_types_and_initalize_woodwork(
-            X_ww.ww.logical_types, X_t_df
-        )
+        X_t_df.ww.init(schema=X_ww.ww.schema)
```

Comment:
I don't think we should do this. Since this is the base class, there's no way to know how the transformer is modifying the features. X_t_df could have fewer columns than X_w, which could result in an error or if they have the same columns the types could be getting converted.

I added `assert False` before line 53 and 72 in this file and ran `pytest -n 8 evalml/tests/component_tests`. Only one test failed and it's because a `MockTransformer` relied on the implementation of `transform` in the base class.

That tells me that we can get rid of `transform` in the base class because every component we test (except a mock one used in tests) implements `transform`. We should mark that as an abstract method. The other thing that tells me is that all components we test rely on `self.fit(X, y).transform(X, y)` since none of the tests triggered the assertion in `fit_transform`. We should get rid of the `try` in `fit_transform`.